### PR TITLE
[sns-quill] Bump to the latest mono-repo version for the SNS-1 release

### PR DESCRIPTION
**Pull request by DanielThurau**
## Summary
I've bumped the Cargo.toml versions of the ic monorepo to `1fc0208b9aeed0554b1be2711605e5b54ace9d6a` which is the version we release for SNS-1. I've also changed the candid to match our production versions, and addressed downstream changes (like handling the new icrc1 API and using the right Principal with ic-agent).

I also updated the pulbic.rs command to generate what an SNS neuron Id is. This will help tremendously when debugging. To use the command you just specify a memo flag.
```
$ sns-quill --canister-ids-file canister_ids.json --pem-file  ~/code/sns-1/sns-1-developer.pem public-ids --memo 5                       
Principal id: 73his-v347r-keigq-mah4l-4sruk-4by32-iehd4-waept-meslh-ewxzn-aqe
NNS account id: 94ab58bdd793cfe883e8452fb58316cc54118d0a9ffbec67a0e2f6ba10d17703
SNS neuron id (memo = 5): 962fa4c7a97a61e961a97139133932b42874e4bc64282cb9f57657b7a32b06cb
```

## Tests
- all 24 tests pass with the new API changes
- I've spun up an SNS-1 on a testnet and tests all commands against that instance.

### CODE REVIEW

**Review by rstout — APPROVED** [review]

File: src/commands/make_upgrade_canister_proposal.rs [hunk 1]
```diff
@@ -8,12 +8,12 @@ use crate::{
 };
 use anyhow::{Context, Error};
 use candid::Encode;
+use candid::Principal;
```
Comment (max-dfinity) [inline src/commands/make_upgrade_canister_proposal.rs:11]: is this the same type?

File: src/commands/public.rs [hunk 2]
```diff
@@ -1,35 +1,70 @@
...
-    match opts.principal_id {
-        Some(principal_id) => {
-            let principal_id = ic_types::Principal::from_text(principal_id)?;
-            Ok((principal_id, get_account_id(principal_id)?))
+    opts: &PublicOpts,
+) -> AnyhowResult<(Principal, AccountIdentifier, Option<NeuronId>)> {
+    match (opts.principal_id.as_ref(), opts.memo) {
+        (Some(principal_id), None) => {
+            let principal_id = Principal::from_text(principal_id)?;
+            Ok((principal_id, get_account_id(principal_id)?, None))
```
Comment (max-dfinity) [inline src/commands/public.rs]: No need to fix, but all the different cases in this function seem to be doing almost identical things w/ the exception of the first step

File: src/commands/send.rs [hunk 3]
```diff
@@ -149,7 +153,10 @@ impl FromStr for SupportedResponse {
         match input {
             "icrc1_balance_of" => Ok(SupportedResponse::AccountBalance),
             "icrc1_transfer" => Ok(SupportedResponse::Transfer),
+            "list_deployed_snses" => Ok(SupportedResponse::ListDeployedSnses),
             "manage_neuron" => Ok(SupportedResponse::ManageNeuron),
+            "send_dfx" => Ok(SupportedResponse::IcpTransfer),
```
Comment (max-dfinity) [inline src/commands/send.rs:160]: why is this called "send_dfx" if it's IcpTransfer?

File: src/commands/make_upgrade_canister_proposal.rs (hunk 1 above)
Comment (DanielThurau) [reply src/commands/make_upgrade_canister_proposal.rs:11]: Yes, and the required type of ic-agent

File: src/commands/send.rs (hunk 3 above)
Comment (DanielThurau) [reply src/commands/send.rs:160]: Legacy reasons that I didn't want to mess with in this particular MR. This is the method that is supported in the current checked in icp_ledger.did and used to transfer ICP around. I'd like to use the ic monorepo `transfer` as that's more up to date but `send_dfx` is still supported and what Adam included when writing the Swap command. I didn't want to take on too much in the PR. As for the naming choices: the `send_dfx` is what is matched on, but I thought a better name for `SupportedResponse` would be `IcpTransfer`.

File: src/commands/public.rs (hunk 2 above)
Comment (DanielThurau) [reply src/commands/public.rs]: Cleaned up in next revision.

File: src/commands/send.rs (hunk 3 above)
Comment (max-dfinity) [reply src/commands/send.rs:160]: Sounds good.  If it's legacy, that's enough reason to leave it alone for now.  But agree it'd be nice to fix later

**Review by max-dfinity — APPROVED** [review]

File: Cargo.lock [hunk 4]
```diff
@@ -1668,43 +1642,35 @@ dependencies = [
...
-name = "ic-cdk"
-version = "0.5.5"
+version = "0.6.4"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "141341c2cb26c4e5f196d5b64305c677b1abebe093eefbe2c622b4a2e6de8a9f"
+checksum = "d12a3ef5de121c699a4db746067d7810c68446804545dd9ecc39f05b6e4c5d74"
 dependencies = [
  "candid",
  "cfg-if 1.0.0",
+ "ic0",
```
Comment (daniel-wong-dfinity-org) [inline Cargo.lock:1662]: Looks suspicious. This might be why the linker is looking for the definition of things like canister_self_size.

**Pull request merged**